### ISSUE TITLE
Support an `actor` mode for pipelines

### ISSUE THREAD

**Issue by thewtex** [InsightSoftwareConsortium/itk-wasm#760]
This mode would enable working with pipelines following the [Actor Model](https://en.wikipedia.org/wiki/Actor_model) in addition to use as a pure function.

For Emscripten:

- Explicitly export `stackSave`, `stackRestore`, per https://github.com/emscripten-core/emscripten/commit/cc1112ad32d13a67fe2534a4e47e3177e736f95b
- Add an `actor: bool` to `runPipeline`'s options.
- Call `stackSave` / `stackPtr` per   https://github.com/Kitware/VolView/pull/281/commits/7690cde3c3d12bf78fd096b7a5b604d55d5fdc96

For WASI:

- And an `actor: boolean` option to `run_pipeline`
- Return an `actor` for re-use -- the last positional output? The wasm runtime `store`?
- Do not run destructor's at the end of execution.
- Run `stackSave` and `stackRestore`?

**Comment by floryst** [comment]
Should we expect pipelines operating in such fashion to call into `main()`? Or should we ccall into some other entrypoint we define? The former handles allocating args, while I'm not sure about the latter.

**Comment by thewtex** [comment]
Yes, actors can allocate their own internal memory and manage their own internal state. For reasons of efficiency and safety, inputs and outputs are allocated, "lowered" and "lifted" by itk-wasm bindgen or, eventually, wit bindgen.

**Comment by floryst** [comment]
Update: I encountered the stack leak behavior in itk-wasm@1.0.0-b.141 when calling readImageBlob many times using the same webworker. I fixed it by patching the itk-image-io files to call stackSave/stackRestore inside `callMain()`. I haven't evaluated this yet in VolView.
